pkg/controller/accountclaim/accountclaim_controller.go

```diff
@@ -136,140 +137,13 @@ func (r *ReconcileAccountClaim) Reconcile(request reconcile.Request) (reconcile.
-			if err != nil {
-				return reconcile.Result{}, err
-			}
-
-			// Set the accountLink of the AccountClaim to the new account if create is successful
-			accountClaim.Spec.AccountLink = newAccount.Name
-			err = r.client.Update(context.TODO(), accountClaim)
-			if err != nil {
-				return reconcile.Result{}, err
-			}
-			// Requeue this claim request in 30 seconds as we need to check to see if the account is ready
-			// so we can update the AccountClaim `status.state` to `true`
-			return reconcile.Result{RequeueAfter: time.Second * waitPeriod}, nil
-		}
-
-		// Get the account and check if its Ready
-		byocAccount := &awsv1alpha1.Account{}
-		err = r.client.Get(context.TODO(), types.NamespacedName{Name: accountClaim.Spec.AccountLink, Namespace: awsv1alpha1.AccountCrNamespace}, byocAccount)
-		if err != nil {
-			return reconcile.Result{}, err
-		}
-
-		if byocAccount.Status.State != string(awsv1alpha1.AccountReady) {
-			if byocAccount.Status.State == string(awsv1alpha1.AccountFailed) {
-				accountClaim.Status.State = awsv1alpha1.ClaimStatusError
-				message := "CCS Account Failed"
-				accountClaim.Status.Conditions = controllerutils.SetAccountClaimCondition(
-					accountClaim.Status.Conditions,
-					awsv1alpha1.CCSAccountClaimFailed,
-					corev1.ConditionTrue,
-					string(awsv1alpha1.CCSAccountClaimFailed),
-					message,
-					controllerutils.UpdateConditionNever,
-					accountClaim.Spec.BYOCAWSAccountID != "",
-				)
-				// Update the status on AccountClaim
-				return reconcile.Result{}, r.statusUpdate(reqLogger, accountClaim)
-			}
-			waitMsg := fmt.Sprintf("%s is not Ready yet, requeuing in %d seconds", byocAccount.Name, waitPeriod)
-			reqLogger.Info(waitMsg, "Account Status", byocAccount.Status.State)
-			return reconcile.Result{RequeueAfter: time.Second * waitPeriod}, nil
-		}
-
-		if byocAccount.Status.State == string(awsv1alpha1.AccountReady) && accountClaim.Status.State != awsv1alpha1.ClaimStatusReady {
-			accountClaim.Status.State = awsv1alpha1.ClaimStatusReady
-			message := "BYOC account ready"
-			accountClaim.Status.Conditions = controllerutils.SetAccountClaimCondition(
-				accountClaim.Status.Conditions,
-				awsv1alpha1.AccountClaimed,
-				corev1.ConditionTrue,
-				AccountClaimed,
-				message,
-				controllerutils.UpdateConditionNever,
-				accountClaim.Spec.BYOCAWSAccountID != "",
-			)
-			// Update the status on AccountClaim
-			return reconcile.Result{}, r.statusUpdate(reqLogger, accountClaim)
-		}
-
-		if !accountClaim.Spec.ManualSTSMode {
-			// Create secret for OCM to consume
-			if !r.checkIAMSecretExists(accountClaim.Spec.AwsCredentialSecret.Name, accountClaim.Spec.AwsCredentialSecret.Namespace) {
-				err = r.createIAMSecret(reqLogger, accountClaim, byocAccount)
-				if err != nil {
-					return reconcile.Result{}, nil
-				}
-			}
-		}
-
-		return reconcile.Result{}, nil
-
+	res, err := r.handleBYOCAccountClaim(reqLogger, accountClaim)
+	if err != nil || res.RequeueAfter > 0 {
+		return res, err
```

Comment:
I think we need to do something more like:

```
if accountClaim.Spec.BYOC {
  return r.handleBYOCAccountClaim(reqLogger, accountClaim)
}
```
in order to properly return, otherwise we're leaking out into the non-ccs methods below.

Reply:
Fixed